### PR TITLE
DOC: in API reference, in long signatures display each parameter on its own line 

### PR DESCRIPTION
**Pull request by TLouf**
## PR summary
In the docs, for long signatures, display every parameter on its own line to improve legibility.

## PR checklist

- [N/A] "closes #0000" is in the body of the PR description to [link the related issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)
- [N/A] new and changed code is [tested](https://matplotlib.org/devdocs/devel/testing.html)
- [N/A] *Plotting related* features are demonstrated in an [example](https://matplotlib.org/devdocs/devel/documenting_mpl.html#writing-examples-and-tutorials)
- [N/A] *New Features* and *API Changes* are noted with a [directive and release note](https://matplotlib.org/devdocs/devel/coding_guide.html#new-features-and-api-changes)
- [N/A] Documentation complies with [general](https://matplotlib.org/devdocs/devel/documenting_mpl.html#writing-rest-pages) and [docstring](https://matplotlib.org/devdocs/devel/documenting_mpl.html#writing-docstrings) guidelines

--------

https://github.com/sphinx-doc/sphinx/pull/11011 introduced a sphinx-native way to display parameters on separate lines, setting the value of [maximum_signature_line_length](https://www.sphinx-doc.org/en/master/usage/configuration.html#confval-maximum_signature_line_length) in conf.py.

Here's a before/after for [`matplotlib.pyplot.subplots`](https://matplotlib.org/stable/api/_as_gen/matplotlib.pyplot.subplots.html)
- Before:
image

- After:
image

And signatures that don't exceed `maximum_signature_line_length` characters would stay the same. Feel free to change the number of characters set by `maximum_signature_line_length`, I just set it to 88 because it's a common convention in python code files, but I have no strong opinion on what would be the "perfect" number.

### CODE REVIEW

**Comment by rcomer** [comment]
Some of our signatures are very long.  For example, with this change the [boxplot](https://output.circle-artifacts.com/output/job/7ae3d237-667c-4324-8945-c0d6382df326/artifacts/0/doc/build/html/api/_as_gen/matplotlib.axes.Axes.boxplot.html#) signature takes up most of my screen, so I have to scroll before I see the function description.  I am not sure that this is desirable.

**Comment by ksunden** [comment]
This may make it more reasonable to add type hints into the rendered html. Not a high priority, as we still have the types listed in docstrings (and have no plans to change that), but the rendering to html being wrapped was one of the major points against including them (see #25590, where I was debating trying to write a sphinx extension to do this)

Note I still do think sphinx being able to source type hints from the pyi stub files (or moving to inline type hints, but that is its own hassle) is a prerequisite for rendering type hints, to avoid pyplot and Axes methods from displaying significantly differently.

It is possible to simply not pin sphinx so high, as the config variable will just be ignored. I think we could pin it (or just rely on `pip install --upgrade` in our doc builds, but _allow_ others to build with older versions... not ideal but _some_ difference should be expected with different sphinx versions, this is just a bit more visible than a lot of them)

That said, sphinx is not a runtime dependency, and not _too_ offended by a relatively narrow pin (though this particular pin was only released 4 days ago, so it is _quite_ aggressive)

I will note that if we are updating the sphinx pin, it should be updated in `requirements/doc/doc-requirements.txt` as well

File: environment.yml [hunk 1]
```diff
@@ -34,7 +34,7 @@ dependencies:
   - packaging
   - pydata-sphinx-theme
   - pyyaml
-  - sphinx>=1.8.1,!=2.0.0
+  - sphinx>=7.1.1
```
Comment (jklymak) [inline environment.yml:37]: I dont think we should be pinning a sphinx that was released four days ago.

**Review by jklymak — COMMENTED** [review]
Agree with @rcomer that having the signature take up many inches of screen space is not ideal, particularly as the parameters are typically explained in detail tabular form below.

File: environment.yml (hunk 1 above)
Comment (tacaswell) [reply environment.yml:37]: https://matplotlib.org/stable/devel/min_dep_policy.html#test-and-documentation-dependencies

> We will support at least minor versions of the development dependencies released in the 12 months prior to our planned release.

Comment (tacaswell) [reply environment.yml:37]: The highest we can pin this is ~ 5.0.0

**Review by tacaswell — CHANGES_REQUESTED** [review]
Please reduce the minimum version of sphinx to be at least 12 months old.

**Comment by timhoffm** [comment]
It's good to know that this is now possible. But for now I'm -0.5 on this. I'm following the argument that this can fill large parts of a screen and that the paramters section already has a list-like structure.

We should definitively revisit if we want to add inline annotations here (but as said there are no plans for this yet).

Note also that future versions of pydata-sphinx-theme will render default values more muted, which already improves the readability of long parameter lists: https://github.com/pydata/pydata-sphinx-theme/pull/1323

**Pull request closed without merging**